toolchain/check/interface_support.h

```diff
@@ -0,0 +1,45 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#ifndef CARBON_TOOLCHAIN_CHECK_INTERFACE_SUPPORT_H_
+#define CARBON_TOOLCHAIN_CHECK_INTERFACE_SUPPORT_H_
+
+#include <optional>
+
+#include "toolchain/check/context.h"
+#include "toolchain/check/decl_name_stack.h"
+#include "toolchain/check/name_component.h"
+#include "toolchain/parse/node_ids.h"
+#include "toolchain/sem_ir/entity_with_params_base.h"
+#include "toolchain/sem_ir/name_scope.h"
+#include "toolchain/sem_ir/typed_insts.h"
+
+namespace Carbon::Check {
+
+// Given a search result for `name_context`, return the previous valid
+// declaration of `name_context` if there is one. Otherwise, produce diagnostics
+// if needed and return nullopt.
```

Comment:
```suggestion
// Returns the previous valid
// declaration of `name_context` if there is one. Otherwise, produces diagnostics
// if needed and return nullopt.
```

nit: see below style note -- suggesting a different edit here since `name_context` is already part of the comment, so "given `name_context`" seems like it can be dropped.

Reply:
Done, but added some more to make the first part more helpful.